pkg/leech/config.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package leech
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/abcxyz/pkg/cfgloader"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/sethvargo/go-envconfig"
+)
+
+// Config defines the set over environment variables required
+// for running this application.
+type Config struct {
+	BatchSize        int    `env:"BATCH_SIZE,required,default=100"`
+	EventsProjectID  string `env:"EVENTS_PROJECT_ID,required"`
+	EventsTable      string `env:"EVENTS_TABLE,required"`
+	GitHubAppID      string `env:"GITHUB_APP_ID,required"`
+	GitHubInstallID  string `env:"GITHUB_INSTALL_ID,required"`
+	GitHubPrivateKey string `env:"GITHUB_PRIVATE_KEY,required"`
+	LeechProjectID   string `env:"LEECH_PROJECT_ID,required"`
+	LeechTable       string `env:"LEECH_TABLE,required"`
+	LogsBucketName   string `env:"LOGS_BUCKET_NAME,required"`
+}
+
+// Validate validates the service config after load.
+func (cfg *Config) Validate() error {
+	if cfg.BatchSize <= 0 {
+		return fmt.Errorf("BATCH_SIZE is required to be a positive integer")
+	}
+
+	if cfg.EventsTable == "" {
```

Comment:
Are all these necessary with go-envconfig and the required key above?

Reply:
It's OK to remove `required` from the envconfig tags because you already have the `Validate()` func. We probably just didn't bother to clean them up in other code base.

@verbanicm It's not super relevant in this context. But when we need to load config from multiple sources (vs. just envs), the envconfig `required` falls short because it doesn't know the existence of other sources. That's why `Validate()` is needed and arguably more flexible.

Reply:
I removed the `required` tag.